qiskit/quantum_info/states/_states.py

```diff
@@ -48,7 +48,9 @@ def random_state(num):
     Returns:
         ndarray:  state(2**num) a random quantum state.
     """
-    x = np.random.random(1 << num)+0.00000001
+    # Random array over interval (0, 1]
+    x = np.random.random(1 << num)
+    x += x == 0
```

Comment:
Yes, I wasn't sure how clear my fix was tbh. I do believe it works however. 

My understanding of the problem was that we wanted `x` to be a random array with values in (0, 1]. The original code avoided zero values by adding `0.00000001` to each element but this had the side effect that in rare cases this was enough to make elements become greater than 1.

Rather than try and limit the random numbers in some way I ended up going for flipping any exact zeros generated to become ones. `x == 0` gives an array of bools which are `True` whenever an element is zero and so the addition operation adds `1` to any zero elements but leaves everything else unchanged.

Or have I completely missed the point?